### PR TITLE
Add automated test script for OrbitService's memory watchdog

### PR DESCRIPTION
**Pull request by dpallotti**
Bug: http://b/214164984

Test:
 - Ran the script locally.
 - Created nightly build and ran on the infrastructure:
   - `blaze test`: http://sponge2/e0df3991-3878-4596-a8d2-559cbbe14c9c
   - `guitar run`: http://fusion2/4101aa6d-b637-4c5b-b285-dce21f3e8869

### CODE REVIEW

File: contrib/automation_tests/test_cases/capture_window.py [hunk 1]
```diff
@@ -347,16 +347,19 @@ def _set_capture_options(self, collect_thread_states: bool, collect_system_memor
...
         self.find_control('Button', 'OK', parent=capture_options_dialog).click_input()
 
-    def _take_capture(self, length_in_seconds: int):
+    def _take_capture(self, length_in_seconds: int, expect_interrupted_by_service: bool):
         capture_tab = self.find_control('Group', "CaptureTab")
         toggle_capture_button = self.find_control('Button', 'Toggle Capture', parent=capture_tab)
 
         logging.info('Starting to capture for {} seconds'.format(length_in_seconds))
         toggle_capture_button.click_input()
-        time.sleep(length_in_seconds)
```
Comment (reichlfl) [inline contrib/automation_tests/test_cases/capture_window.py:356]: This line got lost in refactoring and should be included in the `else:` block below

**Review by reichlfl — CHANGES_REQUESTED** [review]
Please check the missing `time.sleep`.
Otherwise I'm fine with merging it as-is, but I would suggest to move the new functionality into a new class test case class instead of adding a parameter to `Capture`, as it is quite a different use-case. Consider a slight refactoring and deriving from `Capture` instead of extending.

File: contrib/automation_tests/test_cases/capture_window.py (hunk 1 above)
Comment (dpallotti) [reply contrib/automation_tests/test_cases/capture_window.py:356]: Whoops.

**Review by danielfenner — APPROVED** [review]

**Review by reichlfl — APPROVED** [review]
LGTM, thanks!

**Pull request merged**